middleware/static/static.go

```diff
@@ -0,0 +1,170 @@
+package static
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/gofiber/fiber/v3"
+	"github.com/gofiber/utils/v2"
+	"github.com/valyala/fasthttp"
+)
+
+// New creates a new middleware handler.
+// The root argument specifies the root directory from which to serve static assets.
+//
+// Note: Root has to be string or fs.FS, otherwise it will panic.
+func New(root string, cfg ...Config) fiber.Handler {
```

Comment:
Is something else possible here?
the use of a union in golang ? or with other possibilities?

```go
func NewFromRootString(root string, cfg ...Config) fiber.Handler {
	return newMiddlewareHandler(root, nil, cfg...)
}

// NewFromRootFS creates a new middleware handler with an fs.FS root.
func NewFromRootFS(root fs.FS, cfg ...Config) fiber.Handler {
	return newMiddlewareHandler("", root, cfg...)
}
```

or

```go
func New(root interface{}, cfg ...Config) fiber.Handler {
	// Determine root type
	var rootStr string
	var rootFS fs.FS

	switch v := root.(type) {
	case string:
		rootStr = v
	case fs.FS:
		rootFS = v
		rootStr = "." // Default rootStr to current directory for fs.FS
	default:
		panic("static: root must be a string or fs.FS")
	}
```

and please give examples of use for both usecases (filesystem&path) in the markdown

Reply:
since you can use the root node together with the embedded filesystem as a prefix, we leave the structure like this